File: Prácticas/P3/Java/civitas/src/JuegoTexto/Controlador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package JuegoTexto;


import civitas.CivitasJuego;
import civitas.OperacionesInmobiliarias;
import civitas.OperacionesJuego;
import civitas.Respuestas;
import civitas.SalidasCarcel;
import civitas.OperacionInmobiliaria;

/**
 *
 * @author Jose Padial
 */
class Controlador {
    private CivitasJuego juego;
    private VistaTextual vista;
    
    Controlador(CivitasJuego juego, VistaTextual vista){
        this.juego=juego;
        this.vista=vista;
    }
    
    void juega(){
        vista.setCivitasJuego(juego);
        while(!juego.finalDelJuego()){
            System.out.println("");
            vista.actualizarVista();
            vista.pausa();
            OperacionesJuego operacion = juego.siguientePaso();
            vista.mostrarSiguienteOperacion(operacion);
            if(operacion!=OperacionesJuego.PASAR_TURNO){
                vista.mostrarEventos();
            }
            if(!juego.finalDelJuego()){
                switch(operacion){
                    case COMPRAR:
                        Respuestas respuesta=vista.comprar();
                        if(respuesta==Respuestas.SI){
                            juego.comprar();
                        }
                        juego.siguientePasoCompletado(operacion);
                        break;
                    
                    case GESTIONAR:
                        vista.gestionar();
                        int igest=vista.getGestion();
                        int ip=vista.getPropiedad();
                        OperacionInmobiliaria op=new OperacionInmobiliaria(OperacionesInmobiliarias.values()[igest], ip);
                        switch(OperacionesInmobiliarias.values()[igest]){
                            case VENDER:
                                juego.vender(ip);
                                break;
                            case HIPOTECAR:
                                juego.hipotecar(ip);
                                break;
                            case CANCELAR_HIPOTECA:
                                juego.cancelarHipoteca(ip);
                                break;
                            case CONSTRUIR_CASA:
                                juego.construirCasa(ip);
                                break;
                            case CONSTRUIR_HOTEL:
                                juego.construirHotel(ip);
                                break;
                            case TERMINAR:
                                juego.siguientePasoCompletado(operacion);
                                break;
                        }
                        break;
                    case SALIR_CARCEL:
                        SalidasCarcel salida= vista.salirCarcel();
                        if(salida==SalidasCarcel.PAGANDO)
                            juego.salirCarcelPagando();
                        else
                            juego.salirCarcelTirando();
                        juego.siguientePasoCompletado(operacion);
                        break;
                    }
            }
        }
        juego.ranking();
    }
}

File: Prácticas/P3/Ruby/civitas/lib/controlador.rb
# encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "civitas_juego.rb"
require_relative "operacion_inmobiliaria.rb"
require_relative "operaciones_juego.rb"
require_relative "respuestas.rb"
require_relative "operaciones_inmobiliarias.rb"
require_relative "salidas_carcel.rb"

module JuegoTexto
  class Controlador
    include Civitas
    def initialize(juego,vista)
      @juego = juego
      @vista = vista
    end
    
    def juega
      @vista.setCivitasJuego(@juego)
      while !@juego.final_del_juego
        @vista.actualizarVista
        @vista.pausa
        @operacion = @juego.siguiente_paso
        @vista.mostrarSiguienteOperacion(@operacion)
        if @operacion != OperacionesJuego::PASARTURNO
          @vista.mostrarEventos
        end
        if !@juego.final_del_juego
          case @operacion
          when OperacionesJuego::COMPRAR
            @respuesta = @vista.comprar
            if @respuesta == Respuesta::SI
              @juego.comprar
            end
            @juego.siguiente_paso_completado(@operacion)
          when OperacionesJuego::GESTIONAR
            @vista.gestionar
            @igest = @vista.i_gestion
            @ip = @vista.i_propiedad
            lista_respuestas = [OperacionesInmobiliarias::VENDER,OperacionesInmobiliarias::HIPOTECAR,OperacionesInmobiliarias::CANCELAR_HIPOTECA,OperacionesInmobiliarias::CONSTRUIR_CASA,OperacionesInmobiliarias::CONSTRUIR_HOTEL,OperacionesInmobiliarias::TERMINAR]
            @op = OperacionInmobiliaria.new(lista_respuestas[@igest],@ip)
            case lista_respuestas[@igest]
            when OperacionInmobiliaria::VENDER
              @juego.vender(@ip)
            when OperacionInmobiliaria::HIPOTECAR
              @juego.hipotecar(@ip)
            when OperacionInmobiliaria::CANCELARHIPOTECA
              @juego.cancelar_hipoteca(@ip)
            when OperacionInmobiliaria::CONSTRUIRCASA
              @juego.construir_casa(@ip)
            when OperacionInmobiliaria::CONSTRUIRHOTEL
              @juego.construir_hotel(@ip)
            when OperacionInmobiliaria::TERMINAR
              @juego.siguiente_paso_completado(@operacion)
            end
          when OperacionesJuego::SALIRCARCEL
            @salida = @vista.salir_carcel
            if @salida == SalidasCarcel::PAGANDO
              @juego.salir_carcel_pagando
            else
              @juego.salir_carcel_tirando
            end
            @juego.siguiente_paso_completado(@operacion)
          end
        end
      end
      
      @juego.ranking
    end
  end
end
